clippy_lints/src/booleans.rs

```diff
@@ -502,6 +502,8 @@ impl<'a, 'tcx> Visitor<'tcx> for NotSimplificationVisitor<'a, 'tcx> {
             && !inner.span.from_expansion()
             && let Some(suggestion) = simplify_not(self.cx, inner)
             && self.cx.tcx.lint_level_at_node(NONMINIMAL_BOOL, expr.hir_id).0 != Level::Allow
+            && let Some(snippet) = snippet_opt(self.cx, expr.span)
+            && !snippet.contains("assert")
```

Comment:
We don't want to trigger on any macro expansions since you often can't do anything about it, some cases were caught but this one slipped through since the expression called `inner` here is passed into the `assert` macro. We can catch these kinds as well by adding `&& !expr.span.from_expansion()` above `&& !inner.span.from_expansion()`

It would be nice to have a test case added too (in `tests/ui/nonminimal_bool.rs`)